### DISCUSSION TITLE
Subrow column structure

### DISCUSSION THREAD

**Discussion by WildEgo** [TanStack/table#4244] [Q&A] [answered]
Hey!
I was wondering if it's possible for my expandable sub rows to have a different column structure such as:

Main columns:
 - ID
 - Name
 - Actions

Subrow:
 - Message
 
I didn't find anything in the docs so I was wondering if it is possible.

Regards,
Luis Bizarro

**Comment by KevinVandy** [comment, marked as answer]
Yes it is possible. We're about to add a subcomponents example to the docs actually 

Here is the example now in the docs https://tanstack.com/table/v8/docs/examples/react/sub-components

**Reply by WildEgo** [reply]
Thanks mate, I'll have a good look at it, but it seems perfect from what I can see

**Reply by KevinVandy** [reply]
I've made stuff like this with that same pattern too. https://www.material-react-table.dev/?path=/story/features-detail-panel-examples--detail-panel-enabled

**Reply by nlbdotdev** [reply]
> Yes it is possible. We're about to add a subcomponents example to the docs actually
> 
> Here is the example now in the docs https://tanstack.com/table/v8/docs/examples/react/sub-components

Howdy, any idea what happened to this example?

**Reply by TomBinford** [reply]
It seems the docs page is gone but the example is still in the repository ([permalink](https://github.com/TanStack/table/tree/0054eb39992434f1bcf6418f50e347f3f0d5f784/examples/react/sub-components))
And you can point codesandbox at it: 
https://codesandbox.io/embed/github/tanstack/table/tree/main/examples/react/sub-components (HEAD)
https://codesandbox.io/embed/github/tanstack/table/tree/0054eb39992434f1bcf6418f50e347f3f0d5f784/examples/react/sub-components (permalink)

**Reply by TomBinford** [reply]
Here is the permalink to the source of KevinVandy's example: https://github.com/KevinVandy/material-react-table/blob/eb287427f6b1d097717016dd71150752a4965159/packages/material-react-table/stories/features/DetailPanel.stories.tsx#L12
Unfortunately none of these seem to have true rows that would work with the table's sorting and filtering, it's only arbitrary React components.

**Reply by nonchain** [reply]
Hi. please update the url of the example:
it is https://tanstack.com/table/v8/docs/framework/react/examples/sub-components
